__tests__/nc-news.test.js

```diff
@@ -205,52 +205,97 @@ describe("201 - POST - /api/articles/:article_id/comments", () => {
       .expect(201)
       .then((response) => {
         const postedComment = response.body.newComment;
-        postedComment.forEach((comment) => {
-          expect(comment).toEqual(
-            expect.objectContaining({
-              comment_id: expect.any(Number),
-              article_id: expect.any(Number),
-              body: expect.any(String),
-              author: expect.any(String),
-              votes: expect.any(Number),
-              created_at: expect.any(String),
-            })
-          );
-        });
+        expect(postedComment).toEqual(
+          expect.objectContaining({
+            comment_id: expect.any(Number),
+            article_id: expect.any(Number),
+            body: expect.any(String),
+            author: expect.any(String),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+          })
+        );
       });
   });
-  it("should respond with a 404 'not found' error if the username trying to be posted does not currently exist within the users database", () => {
-    const newCommentObj = {
-      username: "checkymarcus",
-      body: "do you like my body?",
-    };
+});
+it("should respond with a 404 'not found' error if the username trying to be posted does not currently exist within the users database", () => {
+  const newCommentObj = {
+    username: "checkymarcus",
+    body: "do you like my body?",
+  };
+  return request(app)
+    .post("/api/articles/2/comments")
+    .send(newCommentObj)
+    .expect(404)
+    .then((response) => {
+      const errorMsg = response.body.msg;
+      expect(errorMsg).toBe("Username not found in database");
+    });
+});
+it("should respond with a 400 Bad Request if the article_id is invalid", () => {
+  return request(app)
+    .post("/api/articles/invalid-id/comments")
+    .expect(400)
+    .then((response) => {
+      const errorMsg = response.body.msg;
+      expect(errorMsg).toBe("Bad Request - Invalid ID");
+    });
+});
+it("should respond with a 404 'not found' error if the ID is a valid number but doesn't exist within the database yet", () => {
+  return request(app)
+    .post("/api/articles/4000/comments")
+    .expect(404)
+    .then((response) => {
+      const errorMsg = response.body.msg;
+      const status = response.status;
+      expect(status).toBe(404);
+      expect(errorMsg).toBe("Not Found");
+    });
+});
+describe("200 - PATCH - /api/articles/:article_id", () => {
+  it("should update the articles vote count by the amount sent", () => {
+    const originalArticle = [
```

Comment:
There is no need for mutation testing. We are not unit testing pure functions but rather integration testing the API. The only place the update would affect (mutate) the original data in where we want it to - in the database our API is using